src/Util/Availability/AvailabilityContextBuilder.php

```diff
@@ -0,0 +1,45 @@
+<?php declare(strict_types=1);
+/*
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Swag\PayPal\Util\Availability;
+
+use Shopware\Core\Checkout\Cart\Cart;
+use Shopware\Core\Content\Product\State;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+
+/**
+ * @internal
+ */
+#[Package('checkout')]
+class AvailabilityContextBuilder
+{
+    public static function buildAvailabilityContext(Cart $cart, SalesChannelContext $salesChannelContext): AvailabilityContext
```

Comment:
First, I would like this to be a service not static, just for extendability reasons. Don't see the benefit of this being static.

Second: How about, we have a `buildFromCart`, `buildFromProduct` and `buildFromOrder` to build the context perfectly every time and not modify things like the `totalAmount` manually?

Reply:
1. I don't agree. I think this is a perfect use case for a static class. It has no state, no dependants and no use for extendibility. It was also extracted from a `private` method to allow us to use it in various locations, not just one class, and therefor marked as `@internal`.

2. I had that, but felt it was... dirty. But I was very on the fence, so I can add it back, not an issue.

Reply:
1. Hmm, ok, we can keep the builder, I still disagree with the static class for the PayLaterAvailabilityChecker.
2. I find that way prettier 😍